### ISSUE TITLE
Minimal supported rust version

### ISSUE THREAD

**Issue by wyfo** [scylladb/scylla-rust-driver#444]
While trying to build the driver with the not-so-old rust version 1.58, I've got this error:
```
   Compiling scylla-macros v0.1.1 (https://github.com/numberly/scylla-rust-driver.git?branch=get_endpoints#9a72382a)
   Compiling dashmap v5.3.3
error[E0658]: use of unstable library feature 'available_parallelism'
  --> /home/alexys/.cargo/registry/src/github.com-1ecc6299db9ec823/dashmap-5.3.3/src/lib.rs:55:6
   |
55 |     (std::thread::available_parallelism().map_or(1, usize::from) * 4).next_power_of_two()
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #74479 <https://github.com/rust-lang/rust/issues/74479> for more information

For more information about this error, try `rustc --explain E0658`
```
Indeed, [*available_parallelism*](https://doc.rust-lang.org/stable/std/thread/fn.available_parallelism.html) seems to have been available only since 1.59, and it's used by the version of dashmap pulled by the driver.

Is there any policy planned about the Minimal Supported Rust Version, like in [tokio](https://github.com/tokio-rs/tokio#supported-rust-versions)? Maybe a badge could be added to the README like in dashmap repository?

**Comment by psarna** [comment]
Yeah, it sounds like a good idea to declare a minimum supported version and stick to it. But I wonder, why would the driver try to use dashmap v5.3.3? We explicitly specify 5.2:
https://github.com/scylladb/scylla-rust-driver/blob/f54ace82ff47d4fcb6c9aef3da9a421adb9e4b65/scylla/Cargo.toml#L36

... and according to semver docs (https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html) and common sense, it should require dashmap in version 5.2.x, not 5.3.x.

And I specifically checked that 5.2 does not yet use available_parallelism, so it should be safe to use with Rust 1.58

**Comment by psarna** [comment]
Ok, got it, after consulting with @piodul I see that 0.x have different rules. We specified dashmap "5.2" which is expected to work fine with anything between 5.2 and 6.0.

dashmap officially states that they support Rust 1.59, so we can either decide that it's our minimal supported version too, or roll back this dependency to something older. Personally I'd go with specifying our minimum supported version to 1.59. Any objections?

**Comment by wyfo** [comment]
I have no objection for 1.59.
By the way, it seems that dashmap is only used for `CachingSession`, so the dependency could also be optional like SSL. But that is not really important in my opinion.

**Comment by Jasperav** [comment]
When https://github.com/scylladb/scylla-rust-driver/pull/358 is merged (not sure why not merged yet), a minimal rust version can be specified in the cargo.toml :)

**Comment by psarna** [comment]
https://github.com/scylladb/scylla-rust-driver/pull/358 was merged and is in 0.4.6, closing
